lib/matplotlib/lines.py

```diff
@@ -159,7 +159,24 @@ def _slice_or_none(in_v, slc):
                     "markevery is specified relative to the axes size, but "
                     "the line does not have a Axes as parent")
             # calc cumulative distance along path (in display coords):
-            disp_coords = affine.transform(tpath.vertices)
+            mask = np.isnan(tpath.vertices)
+            if mask.any():
+                aux = np.empty_like(tpath.vertices)
+                for ic, vals in enumerate(tpath.vertices.T):
+                    ok = ~mask[:, ic]
+                    ii = np.where(ok, np.arange(ok.shape[0]), -1)
+                    i0, i1 = np.nonzero(ok)[0][[0, -1]]
+                    ii[:i0] = i0
+                    ilo = np.maximum.accumulate(ii)
+                    ii[mask[:, ic]] = ok.shape[0] - 1
+                    ii[i1:] = i1
+                    ihi = np.minimum.accumulate(ii[::-1])[::-1]
+                    aux[:, ic] = 0.5 * (vals[ilo] + vals[ihi])
```

Comment:
This part really needs a comment for explanation; I'm not sure it's the simplest way to achieve what you want since I'm not sure what it's trying to do.